### PR TITLE
Accessibility: Add documentation around tooling, auditing and a detailed checklist

### PR DESCRIPTION
**Pull request by umpox**
closes https://github.com/sourcegraph/sourcegraph/issues/31238
closes https://github.com/sourcegraph/sourcegraph/issues/31237

## Description

This PR adds:
- High level documentation on how to conduct an accessibility audit
- Instructions on how to use a screen reader
- Instructions on how to use ARC Toolkit to find accessibility issues
- Detailed checklist on auditing different parts of a journey for accessibility issues (e.g. tables, forms, images, etc)

## Test plan

Documentation. Tested by comparing against Sourcegraph.com and the WCAG guidelines.

## How to trial this process

1. Select a small user journey to trial this process on (e.g. Creating a new code monitor)
2. Open [How to conduct an accessibility audit](https://github.com/sourcegraph/sourcegraph/blob/tr/accessibility-guide/doc/dev/background-information/web/accessibility/how-to-audit.md)
3. Follow the docs. If anything seems confusing/inefficient/wrong - please raise on the PR!

### CODE REVIEW

File: doc/dev/background-information/web/accessibility/detailed-checklist.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+- All navigation actions are recognizable as links.
+- All navigation actions are correct focus styles.
+
+**How to test:**
+
+- Run an audit in ARC Toolkit. Select the "Links" test group. It will visually identify each link on the page and raise any warnings. Check that everything that *should* be a link *is* a link.
+
+**Tips**
+
+- Use the [Wildcard `<Link />` component](https://storybook.sgdev.org/?path=/story/wildcard-link--simple).
```
Comment (taylorsperry) [inline doc/dev/background-information/web/accessibility/detailed-checklist.md:100]: 👏 Very cool to see Wildcard components making Accessibility improvements easier!

File: doc/dev/background-information/web/accessibility/how-to-audit.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+3. Navigate through the user journey using a viewport that has a **width of 320px**.
+    - Would a user be able to sufficiently read all required content in the journey?
+    - Would a user be able to correctly and predictably perform each important **action**?
+      - Keep in mind that it is typically harder to select small buttons and icons using a touch device.
+    - **Note:** You don't need to use a physical mobile device to test this. Most browsers support simulating a mobile viewport - just ensure it is set to **320px**.
+      - [Chrome documentation](https://developer.chrome.com/docs/devtools/device-mode/#viewport)
+      - [Firefox documentation](https://developer.mozilla.org/en-US/docs/Tools/Responsive_Design_Mode)
+4. Work through relevant sections from the [detailed checklist](detailed-checklist.md) and ensure that there are no issues.
+
+## Raising an bug
```
Comment (taylorsperry) [inline doc/dev/background-information/web/accessibility/how-to-audit.md:32]: Typo? "a bug" > "an bug"

File: doc/dev/background-information/web/accessibility/how-to-audit.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+If the bug is very small, and you are confident you can quickly fix it - then go ahead and make a PR! If you aren't able to immediately address the bug, then we should create a new GitHub issue using the following steps:
+
```
Comment (taylorsperry) [inline doc/dev/background-information/web/accessibility/how-to-audit.md:35]: "we should" or "you should"?

File: doc/dev/background-information/web/accessibility/how-to-audit.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+1. Open [this GitHub Issue template](https://github.com/sourcegraph/sourcegraph/issues/new?assignees=&labels=accessibility%2Cwcag%2F2.1%2Cwcag%2F2.1%2Ffixing%2Cestimate%2F3d&template=accessibility_issue.yaml&title=%5BAccessibility%5D%3A+)
+    - Provide as much detail as possible about the bug you found, and the behavior you expected to happen.
+    - Make sure that anyone reading the issue would be able to reproduce the behavior. Screenshots, URLs or videos are helpful!
+2. The issue will be automatically added to the [WCAG 2.1 Tracking Issue](https://github.com/sourcegraph/sourcegraph/issues/31476) and the [Accessibility project](https://github.com/orgs/sourcegraph/projects/238) on GitHub.
+3. If your team has capacity to address the issue, then please assign yourself to it.
+    - The Frontend Platform team will triage any unassigned issues and either fix them or assign them to one of our external contractors.
```
Comment (taylorsperry) [inline doc/dev/background-information/web/accessibility/how-to-audit.md:41]: Nice.

**Review by taylorsperry — APPROVED** [review]

**Comment by umpox** [comment]
Merging now, have tested myself and asked others to help with trialing it. Will iterate if required!

**Pull request merged**